src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -80,21 +80,19 @@ public function __construct($name, $value = null, $expire = 0, $path = '/', $dom
      */
     public function __toString()
     {
-        $str = urlencode($this->getName()).'=';
+        $str = ($this->isRaw() ? $this->getName() : urlencode($this->getName())).'=';
 
         if ('' === (string) $this->getValue()) {
             $str .= 'deleted; expires='.gmdate('D, d-M-Y H:i:s T', time() - 31536001);
         } else {
-            $str .= urlencode($this->getValue());
+            $str .= $this->isRaw() ? $this->getValue() : urlencode($this->getValue());
 
             if ($this->getExpiresTime() !== 0) {
                 $str .= '; expires='.gmdate('D, d-M-Y H:i:s T', $this->getExpiresTime());
             }
         }
 
-        if ($this->path) {
-            $str .= '; path='.$this->path;
-        }
+        $str .= '; path='.$this->getPath() ?: '/';
```

Comment:
See https://github.com/symfony/symfony/pull/20569#discussion_r92156772 for my reasoning behind this. Im also fine with `if ($this->getPath()) { ... }` in case the user breaks the [contract](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Cookie.php#L172) in a subclass.